### PR TITLE
refactor: move pool types to dex lib

### PR DESCRIPTION
**Pull request by piavgh**

### CODE REVIEW

File: pkg/pooltypes/pooltypes.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		DodoClassical:              dodo.PoolTypeDodoClassical,
+		DodoVendingMachine:         dodo.PoolTypeDodoVendingMachine,
+		DodoStable:                 dodo.PoolTypeDodoStable,
+		DodoPrivate:                dodo.PoolTypeDodoPrivate,
+		Velodrome:                  velodrome.DexTypeVelodrome,
+		VelodromeV2:                velodromev2.DexTypeVelodromeV2,
+		Velocimeter:                velocimeter.DexTypeVelocimeter,
+		Pearl:                      pearl.DexTypePearl,
+		Ramses:                     ramses.DexTypeRamses,
+		RamsesV2:                   ramses.DexTypeRamses,
```
Comment (it4rb) [inline pkg/pooltypes/pooltypes.go:178]: is this intentional (ramsesv2 has same value as ramses)?

File: pkg/pooltypes/pooltypes.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+		UniswapV2:                  uniswapv2.DexType,
+		QuickPerps:                 quickperps.DexTypeQuickperps,
+		BalancerV1:                 balancerv1.DexType,
+		BalancerV2Weighted:         balancerv2weighted.DexType,
+		BalancerV2Stable:           balancerv2stable.DexType,
+		BalancerV2ComposableStable: balancerv2composablestable.DexType,
+		VelocoreV2CPMM:             velocorev2cpmm.DexType,
+		VelocoreV2WombatStable:     velocorev2wombatstable.DexType,
+		Fulcrom:                    fulcrom.DexTypeFulcrom,
+		SolidlyV3:                  solidlyv3.DexTypeSolidlyV3,
```
Comment (it4rb) [inline pkg/pooltypes/pooltypes.go:227]: there are some values in router-service that are missing here, could you please check them again:
crowdswap-v2
firebird
balancer-weighted
balancer-stable
balancer-meta-stable
balancer-composable-stable
aerodrome
thena
pearl-v2

Comment (piavgh) [reply pkg/pooltypes/pooltypes.go]: `crowdswap-v2`, `aerodrome`, `thena`  and `pearl-v2` are exchanges, not pool types. I double checked and remove them

`firebird` is an exchange that we removed before, so we don't support it

For 4 balancer types, they are legacy and will be removed in the next release on production, but I still keep them with `Legacy` prefix https://github.com/KyberNetwork/kyberswap-dex-lib/pull/276/commits/c76c277f2f6a551d224d8ae935cabe5ff5c423a2#diff-f3b4b36dc05b3652e22697f889804ad47e6220cd2cb6d3f5af6708e8701d3547R149-R152

File: pkg/pooltypes/pooltypes.go (hunk 1 above)
Comment (piavgh) [reply pkg/pooltypes/pooltypes.go]: thanks, I fixed it

**Review by it4rb — APPROVED** [review]

**Pull request merged**